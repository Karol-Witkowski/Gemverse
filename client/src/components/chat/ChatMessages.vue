<template>
  <v-main class="mb-4 pl-0 pt-4">
    <v-list
      class="scrollBar overflow-y-auto"
      color="rgb(248, 248, 248)"
      :key="message._id"
      three-line
      v-for="message in messages"
    >
        <v-list-item :key="message._id">
          <v-list-item-avatar class="py-0 my-0">
            <img :src="`data:image/svg+xml;utf8,${generateAvatar(message.user._id)}`" />
          </v-list-item-avatar>
          <v-list-item-content>
            <v-list-item-title>
              <h5>{{ message.user.username }}</h5>
            </v-list-item-title>
            <v-list-item-subtitle>
              {{ message.message }}
            </v-list-item-subtitle>
          </v-list-item-content>
        </v-list-item>
      <v-divider class="msgDivider" />
    </v-list>
  </v-main>
</template>

<script>
import { generateFromString } from 'generate-avatar';
import { mapGetters } from 'vuex';

export default {
  name: 'ChatMessages',
  props: {
    messages: {},
  },
  data() {
    return {
      drawerToggle: false,
    };
  },

  computed: {
    ...mapGetters(['getCurrentRoom']),
  },

  methods: {
    generateAvatar(id) {
      return generateFromString(id);
    },
  },
};
</script>
<style lang="scss">
.v-list-item__content {
  align-items: initial;
  padding-top: 0px;
}

.v-list:last-of-type > hr {
  display: none;
}

.scrollBar {
  max-height: 31em;
}
</style>
