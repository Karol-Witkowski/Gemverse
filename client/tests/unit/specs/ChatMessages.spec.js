import { createLocalVue, mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';
import ChatMessages from '@/components/chat/ChatMessages.vue';

const localVue = createLocalVue();
let vuetify;
let wrapper;

vuetify = new Vuetify();

describe('Implementation test for ChatMessages.vue', () => {
  beforeEach(() => {
    wrapper = mount(ChatMessages, {
      localVue,
      mocks: {
        $store: {
          getters: {
            getCurrentRoom: {
              _id: '123testid',
              slug: 'room-name',
            },
          },
        },
      },
      vuetify,
      propsData: {
        messages: {},
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('Render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  /* it('Initializes with correct elements', () => {
    // Test buttons initial state
    expect(wrapper.findAll('button').length).toEqual(2);
    expect(wrapper.findAll('button').at(0).text()).toMatch('cancel');
    expect(wrapper.findAll('button').at(1).text()).toMatch('send');

    // Test input initial state
    expect(wrapper.findAll('.v-text-field').length).toEqual(1);
    expect(wrapper.findAll('input').at(0).text()).toEqual('');

    // Test validation initial state
    expect(wrapper.findAll('.v-messages').length).toEqual(1);
    expect(wrapper.findAll('.v-messages').at(0).text()).toEqual('');
    expect(wrapper.vm.isformValid).toBeFalsy();
  }); */
});
