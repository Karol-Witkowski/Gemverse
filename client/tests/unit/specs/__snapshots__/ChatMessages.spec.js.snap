// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Implementation test for ChatMessages.vue Render correctly 1`] = `
<main class="v-main mb-4 pl-4 pt-2" style="padding: 0px 0px 0px 0px;">
  <div class="v-main__wrap">
    <div class="chat ma-0 overflow-x-auto"></div>
  </div>
</main>
`;
